### ISSUE TITLE
Spare z3 from ambiguous record field names 

### ISSUE THREAD

**Issue by rixed** [rixed/ramen#760]
When the same fields appear in several records of different lengths (such as capture_begin and many others that are present everywhere!) we have a "or"  expression trying all these possibilities.
The actual type is asserted to a single record elsewhere, but apparently that's enough to make z3 about twice slower when minimizing type cost.
So maybe we should do away with that whole OR expression whenever the type is already known (which is the case for input/output records).

In other words, each time we have `(assert (= t12345 record38))` do not also emit any `(assert (or (= t12345 whatever)))`
